frontend/app/services/user-service.server.ts

```diff
@@ -0,0 +1,57 @@
+import moize from 'moize';
+import { z } from 'zod';
+
+import { getAuditService } from '~/services/audit-service.server';
+import { getInstrumentationService } from '~/services/instrumentation-service.server';
+import { getEnv } from '~/utils/env.server';
+import { getLogger } from '~/utils/logging.server';
+
+const log = getLogger('user-service.server');
+
+const userCreateInfoSchema = z.object({
+  email: z.string(),
+  userAttributes: z.array(
+    z.object({
+      name: z.string(),
+      value: z.string(),
+    }),
+  ),
+});
+type UserCreateInfo = z.infer<typeof userCreateInfoSchema>;
+
+export const getUserService = moize(createUserService, { onCacheAdd: () => log.info('Creating new user service') });
+
+function createUserService() {
+  const { CDCP_API_BASE_URI } = getEnv();
+
+  async function createUserWithAttributesAndEmail(user: UserCreateInfo, userId: string) {
```

Comment:
We can simplify this function by accepting `userId` and `email` then using these arguments to craft the request body to send to the API. Since the front-end routes will be passing the RAOIDC user ID, the routes won't have to deal creating an array of `userAttributes`.

Reply:
method changed to take an email and userId